test/e2e/persistentips_test.go

```diff
@@ -112,8 +128,7 @@ var _ = Describe("Persistent IPs", func() {
 					WithTimeout(5 * time.Minute).
 					Should(testenv.ContainConditionVMIReady())
 
-				Expect(testenv.ThisVMI(vmi)()).Should(testenv.MatchIPsAtInterfaceByName(networkInterfaceName, ConsistOf(vmiIPsBeforeMigration)))
-
+				Expect(testenv.ThisVMI(vmi)()).Should(testenv.MatchIPs(params.getIPsFunc, ConsistOf(vmiIPsBeforeMigration)))
```

Comment:
I think the reason this convoluted way of checking (original code):
```
Expect(testenv.ThisVMI(vmi)()).Should(testenv.MatchIPs(getIPsFunc, ConsistOf(vmiIPsBeforeMigration)))
```
is done so that we won't have to re-get the VMI instance.
Changing it to what you want needs to look like this:
```
Expect(testenv.Client.Get(context.Background(), client.ObjectKeyFromObject(vmi), vmi)).To(Succeed())
Expect(testenv.VMIPs(vmi)()).To(ConsistOf(vmiIPsBeforeMigration))
```
otherwise the VMI object in the test is not refreshed with the post-migration entity.

would you like this PR to include this refactorment? I kinda feel it should have a separate PR (which is why I didn't refactor it here) - but I don't mind doing it on this PR.

Reply:
OK, I hate how this looks now, but this can be addressed in a follow up. 

Just saying: this looks a lot more complex than it should be. 

Hiding re-fetching the VM behind a function so I can save an eventually, is also not good IMHO.

Reply:
+1. Will tackle in follow up PR